common/src/main/java/com/wynntils/functions/CharacterFunctions.java

```diff
@@ -188,6 +189,38 @@ public List<Class<? extends Model>> getModelDependencies() {
         }
     }
 
+    public static class MoneyFunction extends Function<Integer> {
```

Comment:
I am not fond of this. At first sight this looks sooooo innocent. But!, what you do here is parsing every item in your container and summing up emerald count. That is LOTS of allocations. Not to even mention that if we use multiple of these functions, we do this multiple times. 

So, how do we fix this? Simple. Create a new model, `PlayerInventoryModel`. Parse emerald count there and cache it. You can use `ContainerSetContent` and `SetSlotEvent` to know when to update your cache, and `WorldStateChangeEvent` to know when to reset the cache. The functions can depend on the model (which can even parse emerald count on init, so there is no need to relog to make it work). When you done this, you can go ahead, and use this cache in `InventoryEmeraldCountFeature`, making it much better too! This way, you not only added new functions, but also improved an existing feature.

Reply:
Let me know if you are having problems understanding this. I tried to be as clear as I can, but you can always ask for help :)

Reply:
@kristofbolyai I totally agree with what you are saying about caching this in a model, but I'm just thinking that maybe listening on the container events is not enough? If you run over emeralds and pick them up, will the `ContainerSetContent` event really be sent? I thought that was only if you have your inventory screen open.

So maybe we'll need a new mixin and event to capture when the player picks up items. And that might be a bit too complicated to ask of a newcomer to implement in what was supposed to be a simple PR.

Reply:
Player inventory is a container tok. I think on pickup, wynn always resends the inventory because it reorders stuff etc. Would need testing for sure though.

Reply:
> So maybe we'll need a new mixin and event to capture when the player picks up items. And that might be a bit too complicated to ask of a newcomer to implement in what was supposed to be a simple PR.

I am happy to write the mixin if it turns out to be needed.